src/Slot.ts

```diff
@@ -139,17 +195,40 @@ export function connectSlot<T=void, T2=void>(
      */
 
     // Signature for Slot(<data>) using default param
-    function trigger(data: any): Promise<any>
+    async function trigger(data: any): Promise<any>
 
     // Signature for Slot(<param>, <data>)
-    function trigger(param: string, data: any): Promise<any>
+    async function trigger(param: string, data: any): Promise<any>
 
     // Combined signatures
-    function trigger(firstArg: string | any, secondArg?: any): Promise<any> {
+    async function trigger(firstArg: string | any, secondArg?: any) {
         const data: any = secondArg || firstArg
         const param: string = secondArg ? firstArg : DEFAULT_PARAM
-        const allHandlers = getParamHandlers(param, handlers)
-        return callHandlers(data, allHandlers)
+
+        if (config.noBuffer || transports.length === 0) {
+            const allParamHandlers = getParamHandlers(param, handlers)
+            return callHandlers(data, allParamHandlers)
```

Comment:
Yes this part of the `ts-event-bus` API is unclear. For now I have stuck with the existing. The problem with `promise.resolve` is that you are returning something of type `void` when the slot was configured to return something of some generic type `T`. Never resolving (the behavior of `callHandlers` when no one is listening) solves this.

You will find other similar return type problems for instance with the return type of the `trigger` method: it is typed as any instead of the generic type `T2`, and I believe it is because we sometimes return `T2` and sometimes `T2[]` if there are multiple remote handlers.